packages/protocol/contracts/L1/libs/LibProving.sol

```diff
@@ -183,6 +189,34 @@ library LibProving {
         });
     }
 
+    function overwriteForkChoice(
+        TaikoData.State storage state,
+        TaikoData.Config memory config,
+        AddressResolver resolver,
+        uint256 blockId,
+        TaikoData.BlockEvidence memory evidence
+    ) internal {
+        if (msg.sender != resolver.resolve("forkchoice_failsafe", true)) {
+            revert L1_NO_AUTH_TO_OVERWRITE_FK();
+        }
+
+        if (blockId <= state.lastVerifiedBlockId || blockId >= state.numBlocks) {
```

Comment:
Small but not completely trivial logic that is shared with the prover code, maybe a bit better to move this to a util function called `isBlockBeingProven` or something.